core/src/main/java/org/apache/calcite/plan/RelOptCost.java

```diff
@@ -76,19 +132,28 @@
 
   /**
    * Compares this to another cost.
+   * This is based on the implementation of method {@link #compareCost(RelOptCost)},
+   *  and is not intended to be overridden.
    *
    * @param cost another cost
    * @return true iff this is less than or equal to other cost
    */
-  boolean isLe(RelOptCost cost);
+  default boolean isLe(RelOptCost cost) {
+    ComparisonResult result = compareCost(cost);
+    return result == ComparisonResult.LT || result == ComparisonResult.EQ;
+  }
 
   /**
    * Compares this to another cost.
+   * This is based on the implementation of method {@link #compareCost(RelOptCost)},
+   * and is not intended to be overridden.
    *
    * @param cost another cost
    * @return true iff this is strictly less than other cost
    */
-  boolean isLt(RelOptCost cost);
+  default boolean isLt(RelOptCost cost) {
+    return compareCost(cost) == ComparisonResult.LT;
+  }
```

Comment:
I don't see interface definition needs to be changed, the change should be made in the IMPlementation file.